x/subspaces/keeper/groups.go

```diff
@@ -0,0 +1,113 @@
+	k.AfterSubspaceGroupSaved(ctx, group.SubspaceID, group.ID)
+}
+
+// HasUserGroup returns whether the given subspace has a group with the specified id or not
+func (k Keeper) HasUserGroup(ctx sdk.Context, subspaceID uint64, groupID uint32) bool {
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(types.GroupStoreKey(subspaceID, groupID))
+}
+
+// GetUserGroup returns the group associated with the given id inside the subspace with the provided id.
+// If there is no group associated with the given id the function will return an empty group and false.
+func (k Keeper) GetUserGroup(ctx sdk.Context, subspaceID uint64, groupID uint32) (group types.UserGroup, found bool) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.GroupStoreKey(subspaceID, groupID)
+	if !store.Has(key) {
+		return group, false
+	}
+
+	k.cdc.MustUnmarshal(store.Get(key), &group)
+	return group, true
+}
+
+// DeleteUserGroup deletes the group with the given id from the subspace with the provided id
+func (k Keeper) DeleteUserGroup(ctx sdk.Context, subspaceID uint64, groupID uint32) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Remove all the members from this group
+	var members []sdk.AccAddress
+	k.IterateGroupMembers(ctx, subspaceID, groupID, func(index int64, member sdk.AccAddress) (stop bool) {
+		members = append(members, member)
+		return false
+	})
+
+	for _, member := range members {
+		k.RemoveUserFromGroup(ctx, subspaceID, groupID, member)
+	}
+
+	// Delete the group
+	store.Delete(types.GroupStoreKey(subspaceID, groupID))
+
+	k.Logger(ctx).Info("group deleted", "subspace_id", subspaceID, "group_id", groupID)
+	k.AfterSubspaceGroupDeleted(ctx, subspaceID, groupID)
+}
+
+// --------------------------------------------------------------------------------------------------------------------
+
+// AddUserToGroup adds the given user to the group having the provided id inside the specified subspace.
+// If the group does not exist inside the subspace, it returns an error.
+func (k Keeper) AddUserToGroup(ctx sdk.Context, subspaceID uint64, groupID uint32, user sdk.AccAddress) error {
+	if !k.HasUserGroup(ctx, subspaceID, groupID) {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "group with id %d does not exist", groupID)
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.GroupMemberStoreKey(subspaceID, groupID, user), []byte{0x01})
+
+	k.AfterSubspaceGroupMemberAdded(ctx, subspaceID, groupID, user)
+
+	return nil
+}
+
+// IsMemberOfGroup returns whether the given user is part of the group with
+// the specified id inside the provided subspace
+func (k Keeper) IsMemberOfGroup(ctx sdk.Context, subspaceID uint64, groupID uint32, user sdk.AccAddress) bool {
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(types.GroupMemberStoreKey(subspaceID, groupID, user))
+}
+
+// RemoveUserFromGroup removes the specified user from the subspace group having the given id.
+// If the group does not exist inside the subspace, it returns an error.
```

Comment:
It does not return anything.

Reply:
The documentation is outdated, it's better to not return anything. Let me fix this

```suggestion
```